In [3]:
import pandas as pd
from langdetect import detect

In [8]:
df = pd.read_csv('arceus_reviews.csv')

In [9]:
df.head()

review          date  score  \
0  A fun reinvention of the Pokémon formula, hope...  Jan 29, 2022     10   
1  A fun, challenging (at times), Pokemon game wi...  Jan 30, 2022     10   
2  El juego más infravalorado de la saga y posibl...  Jan 29, 2022     10   
3  Best pokemon game of all time! New pokemon, ne...  Jan 30, 2022     10   
4  One outstanding experience, that is certain. I...   Feb 3, 2022     10   

                             helpful number_of_reviews number_of_ratings  \
0  49 of 62 users found this helpful         2 Reviews         2 Ratings   
1  52 of 72 users found this helpful        60 Reviews       111 Ratings   
2  48 of 62 users found this helpful         8 Reviews        12 Ratings   
3  28 of 35 users found this helpful          1 Review         4 Ratings   
4    4 of 8 users found this helpful          1 Review        14 Ratings   

   average_user_score  
0                 0.0  
1                 5.8  
2                 7.2  
3                 5.0  
4                 9.0

## Unicamente nos es de utilidad las criticas, por lo que solo mantendremos esa columna.

In [10]:
df = df['review'].to_frame()

In [11]:
df.head()

review
0  A fun reinvention of the Pokémon formula, hope...
1  A fun, challenging (at times), Pokemon game wi...
2  El juego más infravalorado de la saga y posibl...
3  Best pokemon game of all time! New pokemon, ne...
4  One outstanding experience, that is certain. I...

In [12]:
len(df)

1067

## Tenemos 1067 criticas.

## Encontremos el lenguaje de cada critica con **langdetect**

In [13]:
df['language'] = df['review'].apply(detect)

In [18]:
df['language'].unique()

array(['en', 'es', 'tr', 'ru', 'pt', 'it', 'id', 'de', 'fr', 'no', 'ja',
       'ko', 'zh-cn'], dtype=object)

In [17]:
print(len(df['language'].unique()))

13


## langdetect encontro 13 idiomas, conservemos unicamente los que están en inglés

In [19]:
df = df[df['language'] == 'en']

In [20]:
df.head()

review language
0  A fun reinvention of the Pokémon formula, hope...       en
1  A fun, challenging (at times), Pokemon game wi...       en
3  Best pokemon game of all time! New pokemon, ne...       en
4  One outstanding experience, that is certain. I...       en
5  Incredible game that moves the Pokemon series ...       en

In [21]:
len(df)

835

## Tenemos 835 críticas en ingles, usaremos estas para entrenar nuestra red.

In [22]:
df.to_csv('eng_reviews.csv')